google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,111 @@
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-compute-engine-oslogin/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-oslogin/master/packaging/debian/control"
+    ],
+    "google-osconfig-agent": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-osconfig-agent/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/osconfig/master/packaging/debian/control"
+    ],
+} # type: archives
+
+def clean_up() -> None:
+    os.remove(UBUNTU_CONTROL_FILENAME)
+    os.remove(GOOGLE_CONTROL_FILENAME)
+
+def parse_control_file(control_file: str | os.PathLike) -> Deb822:
+    with open(control_file) as fp:
+        control_file_text = fp.read()
 
-def parse_control_file(control_file):
-    control_file_text = control_file.read()
     control_file_text_no_blank_lines = os.linesep.join([s for s in control_file_text.splitlines() if s])
     control_fileds = deb822.Deb822(control_file_text_no_blank_lines)
     return control_fileds
 
-
 @click.command()
-@click.option('--upstream-control-file', type=click.File(), required=True, help='Upstream control files to compare')
-@click.option('--ubuntu-control-file', type=click.File(), required=True, help='Ubuntu control files to compare')
-def main(upstream_control_file, ubuntu_control_file):
-    # wget --output-document=ubuntu-control "https://git.launchpad.net/ubuntu/+source/google-guest-agent/tree/debian/control?h=applied/ubuntu/noble-devel"
-    # wget --output-document=ubuntu-control "https://git.launchpad.net/ubuntu/+source/google-guest-agent/plain/debian/control?h=applied/ubuntu/noble-devel"
-    ubuntu_control_fields = parse_control_file(ubuntu_control_file)
-    upstream_control_fields = parse_control_file(upstream_control_file)
-    fields_to_compare = ['Build-Depends', 'Depends', 'Recommends', 'Suggests', 'Breaks', 'Conflicts', 'Replaces', 'Provides']
-
-    # compare the above control fields
+@click.option('--agent-package', required=True,
+              type=click.Choice(["google-guest-agent",
+                                        "gce-compute-image-packages",
+                                        "google-compute-engine-oslogin",
+                                        "google-osconfig-agent"], case_sensitive=False),
+              help='The package to inspect')
+
+def main(agent_package: str) -> None:
+
+    url_list = archive_dict[agent_package]
+
+    ubuntu_req = requests.get(url_list[0])
+    with open(UBUNTU_CONTROL_FILENAME, "w+") as fp:
+        fp.write(ubuntu_req.text)
+
+    google_req = requests.get(url_list[1])
+    with open(GOOGLE_CONTROL_FILENAME, "w+") as fp:
+        fp.write(google_req.text)
+
+    ubuntu_control_fields = parse_control_file(UBUNTU_CONTROL_FILENAME)
+    upstream_control_fields = parse_control_file(GOOGLE_CONTROL_FILENAME)
+
+    fields_to_compare = ['Build-Depends', 'Depends', 'Recommends',
```

Comment:
It might not be a terrible idea to put these in a global "constant" just to make them easier to find and edit in the future, even though this list is only used once.

Reply:
Done, it's in the next upload :smile: